cparser.lua

```diff
@@ -0,0 +1,74 @@
+local luaclang = require "luaclang"
+
+--module table
+local cparser = {}
+
+--visitor function for traversing children of an EnumDecl
+local function enum_child_visitor(cursor, parent, enum_fields)
+        local kind = cursor:getKind()
+        local enum_const
+        if kind == "EnumConstantDecl" then
+                enum_const = {}
+                enum_const.name = cursor:getSpelling()
+        elseif kind == "IntegerLiteral" then
+                enum_fields[#enum_fields].value = parent:getEnumValue()
+        end
+        table.insert(enum_fields, enum_const)
+        return "recurse"
+end
+
+local function enum_handler(cursor, parent, declarations)
+        local enum_decl = {} 
+        local enum_fields = {} 
+        enum_decl.tag = 'enum'
+        enum_decl.name = cursor:getSpelling()
+        cursor:visitChildren(enum_child_visitor, enum_fields)
+        enum_decl.fields = enum_fields
+        table.insert(declarations, enum_decl)
+end
+
+local function function_handler(cursor, parent, declarations)
+        local func_decl = {}
+        local func_params = {}
+        func_decl.tag = 'function'
+        func_decl.name = cursor:getSpelling()
+        func_decl.inline = cursor:isFunctionInlined()
+        func_decl.storage_specifier = cursor:getStorageClass()
+        local type = cursor:getType()
+        local return_type = type:getResultType()
+        func_decl["return"] = return_type:getSpelling()
```

Comment:
I would shorten return to ret just to avoid the clash with the reserved word.
```suggestion
        decl.ret = return_type:getSpelling()
```